datafusion/core/src/physical_plan/sorts/merge.rs

```diff
@@ -37,8 +52,16 @@ pub(crate) fn streaming_merge(
     tracking_metrics: MemTrackingMetrics,
     batch_size: usize,
 ) -> Result<SendableRecordBatchStream> {
-    let streams = RowCursorStream::try_new(schema.as_ref(), expressions, streams)?;
+    if expressions.len() == 1 {
```

Comment:
I think it is worth a comment here explaining the rationale for this for future readers. Something like

```suggestion
    // special case single column primitives to avoid overhead of runtime comparators
    if expressions.len() == 1 {
```